WordPressAuthenticator/Unified Auth/View Related/Site Address/SiteCredentialsViewController.swift

```diff
@@ -541,6 +541,11 @@ extension SiteCredentialsViewController {
         let wporg = WordPressOrgCredentials(username: username, password: password, xmlrpc: xmlrpc, options: options)
         let credentials = AuthenticatorCredentials(wporg: wporg)
 
+        /// If `completionHandler` is available, return early with the credentials.
+        if let completionHandler = completionHandler {
+            return completionHandler(wporg)
```

Comment:
❓ since it early returns, do we not need any of the logic below like `delegate.sync(credentials: credentials)` and its completion block since `configuration.isWPComLoginRequiredForSiteCredentialsLogin` is `false` for WCiOS?

When I looked into the usage of `SiteCredentialsViewController` (not a public class), it always has a non-nil completion block from `WordPressAuthenticator.showSiteCredentialLogin` at the moment. The completion block type is nullable now only for the initial `init?(coder: NSCoder)` if I understand correctly. This means that none of the code below this would be needed 😅 If this is the case, what do you think about deleting the code below so that it's clear for future developers that this is the end of the function?

Reply:
@jaclync  As you can see, the initializer with the non-optional `completionHandler` is only used in `WordPressAuthenticator.showSiteCredentialLogin` - a method to present site credential login directly and triggers the completion block immediately after login completes. The goal is to send the credentials back and not to authenticate. In Woo, this flow is only used during the Jetpack connection flow for users authenticated with WPCom.

`SiteCredentialsViewController` has another initializer  `init?(coder: NSCoder)` without the completion handler, which is used in all the other flows through `SiteCredentialsViewController.instantiate(from: .siteAddress)`. In these flows, we trigger the delegate when the login completes.

Reply:
What do you think if this `finishedLogin` calls two separate functions, one with a completion block and the other without, with `wporg` and `credentials` passed to them from the beginning of the function?

For developers new to this code, I think it can be quite confusing reading about or not knowing these two code paths.

Reply:
Good point! Separate this method into 3 parts in 075d308a039525f119a717641b03d2f74a8815af - lmk what you think!